### PR TITLE
fix `tsh proxy aws --endpoint-url`

### LINKED ISSUE #20798: `tsh proxy aws --endpoint-url` not working
Expected behavior:
`tsh proxy aws --endpoint-url` should work for AWS app access.

Current behavior:
It fails with a signature v4 error:
```
$ aws dynamodb list-tables --region=us-west-2 --endpoint-url https://127.0.0.1:5354
5

An error occurred (InvalidSignatureException) when calling the ListTables operation: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.
```

I traced this issue down to the way we handle signing AWS requests in the agent.
When using `--endpoint-url` we don't start a local AWS forwarder, thus when the HTTP request reaches the AWS handler in the app service agent it has a `Host` header that matches the agent's address instead of AWS. Then we sign the request and modify the Host header before it is ultimately sent to AWS. But `host` is one of the signed headers in the original request. Since we modify the `Host` header *after* signing, it no longer matches the `Host` header that AWS receives, and so sigv4 check fails.

We should either:
1. remove `Host` header before signing so that it is not a sigv4 signed header, or
2. modify `Host` header before signing so the signed `Host` header matches what AWS receives.

I prefer option 2.

Bug details:
- Teleport version: v12.0.0-alpha.1
- Recreation steps:
1. configure an app service agent and an aws console app
4. run `tsh proxy aws --endpoint-url --app <yourapp>`
5. try to connect to the app via `aws` cli

Workaround:
- Use `tsh aws`, which always starts a local AWS forwarder or use `tsh proxy aws` without the `--endpoint-url` flag.

### PR DESCRIPTION
**Pull request by GavinFrazar**
Fixes #20798

This PR modifies the AWS request `Host` header to match the URL `Host` header before signing, to avoid sigv4 failures in AWS. Full description of the problem and this solution are in the issue this PR closes.

The fix is basically a 1 line code change and some comments to explain, the rest of the changes are to make tests fail without the fix. Issue affects v11/v12.

@zmb3 a workaround exists (as described in the issue), so I'm not sure if this should be merged for v12 release or not.

### CODE REVIEW

File: lib/srv/app/aws/handler_test.go [hunk 1]
```diff
@@ -209,17 +212,28 @@ func TestAWSSignerHandler(t *testing.T) {
...
+				assert.Equal(t, tc.wantAuthCredKeyID, awsAuthHeader.KeyID)
+				assert.Equal(t, tc.wantAuthCredService, awsAuthHeader.Service)
+
+				// check that the signature is valid.
+				err = awsutils.VerifyAWSSignature(r, staticAWSCredentials)
+				if !assert.NoError(t, err) {
+					http.Error(w, err.Error(), trace.ErrorToCode(err))
+					return
+				}
+				w.Write(nil)
```
Comment (smallinsky) [inline lib/srv/app/aws/handler_test.go:234]:
```suggestion
				w.WriteHeader(StatusOK) 
```

**Review by smallinsky — APPROVED** [review]
Thanks for the fix. 

@greedy52 
Could you also take a look ? Also I think that is is worth to add the `tsh aws proxy --endpoint-url to our release test plan.

**Review by greedy52 — APPROVED** [review]
👍 Agreed that it's better to sign with `host` header so the host can be verified, for better security.

And agreed we should add `--endpoint-url` to test plan, in addition to `tsh aws --endpoint-url dynamodb list-tables`, I suggest also adding this (if we list sample commands there):
```
tsh aws --endpoint-url ssm start-session --target <target>
```
since `ssm` session (and the ECS equivalent e.g. `execute-command`) only works in `--endpoint-url` mode.

**Review by zmb3 — APPROVED** [review]
LGTM. I don't see any reason why we can't get this in for 12 as long as you verify that everything works as expected.

You just need a `make fix-imports` to appease the linter.

**Pull request merged**